### PR TITLE
Decompile some of flying.s

### PR DESCRIPTION
**Pull request by atasro2**
this file has some sweet and short functions.. inb4 you see the one with over 900 lines

### CODE REVIEW

File: src/battle/anim/flying.c [hunk 1]
```diff
@@ -23,6 +31,288 @@ void sub_80DB458(struct Sprite *sprite);
 void sub_80DB564(struct Sprite *sprite);
 void sub_80DB5E4(struct Sprite *sprite);
 
+void sub_80DA0DC(u8 taskId);
+
+void sub_80DA034(struct Sprite *sprite) 
+{
+    sub_8078764(sprite, 0);
+    sprite->pos1.y += 0x14;
```
Comment (huderlem) [inline src/battle/anim/flying.c]: Use non-hexadecimal numbers for basic operations like this.  In general it's better to reserve hexadecimal for things like bitwise operations and sometimes Ids, such as the sprite palette tags.

File: src/battle/anim/flying.c [hunk 2]
```diff
@@ -23,6 +31,288 @@ void sub_80DB458(struct Sprite *sprite);
...
+    sprite->callback = sub_80DA05C;
+    sub_80DA05C(sprite);
+}
+
+void sub_80DA05C(struct Sprite *sprite)
+{
+    sprite->pos2.x = Sin(sprite->data[1], 0x20);
+    sprite->pos2.y = Cos(sprite->data[1], 0x8);
+    sprite->data[1] = (sprite->data[1] + 5) & 0xFF;
+    ++sprite->data[0];
```
Comment (huderlem) [inline src/battle/anim/flying.c]: This line can be merged with the one below it
```
if (++sprite->data[0] == 71)
```

File: src/battle/anim/flying.c [hunk 3]
```diff
@@ -23,6 +31,288 @@ void sub_80DB458(struct Sprite *sprite);
...
+  gTasks[taskId].data[0] = gBattleAnimArgs[1];
+  gTasks[taskId].data[1] = gBattleAnimArgs[0];
+  gTasks[taskId].data[2] = IndexOfSpritePaletteTag(0x2719);
+  gTasks[taskId].func = sub_80DA0DC;
+}
+
+#ifdef NONMATCHING
+void sub_80DA0DC(u8 taskId)
+{
+    gTasks[taskId].data[10]++;
```
Comment (huderlem) [inline src/battle/anim/flying.c]: same here, can be merged with next line

File: src/battle/anim/flying.c [hunk 4]
```diff
@@ -23,6 +31,288 @@ void sub_80DB458(struct Sprite *sprite);
...
+_080DA164: .4byte 0x00000107\n\
+_080DA168: .4byte 0x00000101\n\
+    .syntax divided\n");
+}
+#endif
+
+void sub_80DA16C(struct Sprite *sprite)
+{
+    InitAnimSpritePos(sprite, 1);
+    if (GetBattlerSide(gAnimBankAttacker) != 0)
```
Comment (huderlem) [inline src/battle/anim/flying.c]: Use the appropriate `B_SIDE` constant here.  I believe `B_SIDE_PLAYER` is `0`.

File: src/battle/anim/flying.c [hunk 5]
```diff
@@ -23,6 +31,288 @@ void sub_80DB458(struct Sprite *sprite);
...
+{
+    if (TranslateAnimSpriteByDeltas(sprite) != 0)
+    {
+        DestroyAnimSprite(sprite);
+    }
+}
+
+void sub_80DA208(struct Sprite *sprite)
+{
+    if (GetBattlerSide(gAnimBankAttacker) != 0)
```
Comment (huderlem) [inline src/battle/anim/flying.c]: `B_SIDE_PLAYER`

File: src/battle/anim/flying.c [hunk 6]
```diff
@@ -23,6 +31,288 @@ void sub_80DB458(struct Sprite *sprite);
...
+            FreeOamMatrix(sprite->oam.matrixNum);
+            sprite->oam.affineMode = 0;
+        }
+        
+        DestroySprite(sprite);
+        gAnimVisualTaskCount--;
+    }
+}
+
 const struct SpriteTemplate gBattleAnimSpriteTemplate_83DA380 =
```
Comment (huderlem) [inline src/battle/anim/flying.c]: Can you move the .rodata above the code?  (.rodata is all of the `const` data). That's how this project is typically organized

**Comment by atasro2** [comment]
done now the code is more optimized (probably not) and its below the .rodata

**Pull request merged**